manifests/charts/istio-cni/templates/configmap-cni.yaml

```diff
@@ -1,34 +1,28 @@
-{{- $defaultBinDir :=
-    (.Capabilities.KubeVersion.GitVersion | contains "-gke") | ternary
-      "/home/kubernetes/bin"
-      "/opt/cni/bin"
-}}
 kind: ConfigMap
 apiVersion: v1
 metadata:
-  name: istio-cni-config
+  name: {{ template "name" . }}-config
   namespace: {{ .Release.Namespace }}
   labels:
-    app: istio-cni
+    app: {{ template "name" . }}
     release: {{ .Release.Name }}
     istio.io/rev: {{ .Values.revision | default "default" }}
     install.operator.istio.io/owning-resource: {{ .Values.ownerName | default "unknown" }}
     operator.istio.io/component: "Cni"
 data:
-  # The CNI network configuration to add to the plugin chain on each node.  The special
-  # values in this config will be automatically populated.
-  cni_network_config: |-
-        {
-          "cniVersion": "0.3.1",
-          "name": "istio-cni",
-          "type": "istio-cni",
-          "log_level": {{ quote .Values.cni.logLevel }},
-          "log_uds_address": "__LOG_UDS_ADDRESS__",
-          {{if .Values.cni.ambient.enabled}}"ambient_enabled": true,{{end}}
-          "cni_event_address": "__CNI_EVENT_ADDRESS__",
-          "kubernetes": {
-              "kubeconfig": "__KUBECONFIG_FILEPATH__",
-              "cni_bin_dir": {{ .Values.cni.cniBinDir | default $defaultBinDir | quote }},
-              "exclude_namespaces": [ {{ range $idx, $ns := .Values.cni.excludeNamespaces }}{{ if $idx }}, {{ end }}{{ quote $ns }}{{ end }} ]
-          }
-        }
+  CURRENT_AGENT_VERSION: {{ .Values.cni.tag | default .Values.global.tag | quote }}
```

Comment:
should we just inline these into the daemonset directly?

Benefit is its consistent with the downward API ones and it auto-rollout if we change

This approach is easier to grok though, so not full sure one is better...

Reply:
> should we just inline these into the daemonset directly?

Yeah I thought about that too - we could. 

Reasons why (for now) I kinda wanna keep the CM:

- Some env vars on the daemonset (the ones that are left) are not overridable. Keeping them separate makes it clearer what we expect to be directly user/values-overridable env vars/config, versus ones that are not.

- For upgrades, we will need a resource with a separate lifecycle.